### ISSUE TITLE
CloudBlobDirectory does not have any public constructor

### ISSUE THREAD

**Issue by Jericho** [Azure/azure-storage-net#639]
The `CloudBlobDirectory` class does not have any public constructor which means that we can't mock this class for unit testing purposes. For instance, the following line of code:
```csharp
var mockBlobDirectory = new Mock<CloudBlobDirectory>(MockBehavior.Strict);
```
causes the following exception:
> System.NotSupportedException : Parent does not have a default constructor. The default constructor must be explicitly defined.

Can you please consider adding a public constructor?

Edit: I'm using `Microsoft.WindowsAzure.Storage, Version=9.0.0.0`.

**Comment by erezvani1529** [comment]
Hi @Jericho ,

We will consider this change to be included in the [Azure Storage Blob package](https://www.nuget.org/packages/Microsoft.Azure.Storage.Blob/9.0.0-preview).

THanks!

**Comment by joopscheer** [comment]
@erezvani1529  Any news on this?

**Comment by richcorless** [comment]
Have just hit this one, has there been any update on this?

**Comment by daniel94104** [comment]
Hit this problem as well today, any updates? Thanks!

**Comment by AustinFelipe** [comment]
Hey guys, it would be great to have news about that!

**Comment by itsrachelle** [comment]
Following. This would be extremely helpful.

**Comment by gonzalolorieto** [comment]
No public ctor AFAIK

**Comment by seanmcc-msft** [comment]
Fixed in version 10.0.3

Commit: https://github.com/Azure/azure-storage-net/commit/550427e7090ab656f4a8279e7cdd85fba6c6cc98
